### PR TITLE
Swift 4 compatibility

### PR DESCRIPTION
**Pull request by jcaracciolo**
## Summary ##
**Issue**: #31
 
This PR migrates the code to swift 4, making it compatible. No implementation changes were done.

### CODE REVIEW

File: NetworkingDemo/Info.plist [hunk 1]
```diff
@@ -2,8 +2,6 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-	<key>UIViewControllerBasedStatusBarAppearance</key>
```
Comment (pablog) [inline NetworkingDemo/Info.plist]: Discard this change

File: NetworkingDemo/Entity.swift [hunk 2]
```diff
@@ -16,7 +16,7 @@ public struct Entity {
     let genre: String
 }
 
-extension Entity: Decodable {
+extension Entity:  Argo.Decodable {
```
Comment (pablog) [inline NetworkingDemo/Entity.swift]: Double space

File: Networking/Extensions/AlamofireExtensions.swift [hunk 3]
```diff
@@ -38,7 +38,7 @@ internal extension Alamofire.DataRequest {
     
     func response() -> SignalProducer<ResponseType, ResponseError> {
         return SignalProducer { observer, disposable in
-            disposable.add { self.task?.cancel() }
+            disposable.observeEnded { self.task?.cancel() }
```
Comment (pablog) [inline Networking/Extensions/AlamofireExtensions.swift]: Are you sure? It sounds weird to me. Do you have a link or something with this?

**Comment by pablog** [comment]
@jcaracciolo Remove all Fastlane stuff from here. 

Then we can think again about it, and decide whether or not to add it, but for sure this pull request "Swift 4 compatibility" is not the right place for these changes.

File: Networking.xcodeproj/xcshareddata/xcschemes/Networking.xcscheme [hunk 4]
```diff
@@ -23,9 +23,10 @@
       </BuildActionEntries>
    </BuildAction>
    <TestAction
-      buildConfiguration = "Debug"
```
Comment (pablog) [inline Networking.xcodeproj/xcshareddata/xcschemes/Networking.xcscheme]: Rollback this. Discard the changes in the whole file.

File: Networking/Extensions/AlamofireExtensions.swift (hunk 3 above)
Comment (jcaracciolo) [reply Networking/Extensions/AlamofireExtensions.swift]: This was a suggestion of Xcode, `'add' has been renamed to 'observeEnded'`: https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Sources/Deprecations%2BRemovals.swift#L21

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 81.68


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-networking-ios/branches/swift-4-compatibility) for more details.

**Pull request merged**